plugins/CircuitViewer/io/CircuitLoader.cpp

```diff
@@ -107,6 +106,52 @@ std::string _getMeshFilenameFromGID(const uint64_t gid,
 }
 #endif
 
+/**
+ * Return a vector of gids.size() elements with the layer index of each gid
+ */
+size_ts _getLayerIds(const brain::Circuit& circuit, const brain::GIDSet& gids)
+{
+    std::vector<brion::GIDSet> layers;
+    try
+    {
+        for (auto layer : {"1", "2", "3", "4", "5", "6"})
+            layers.push_back(circuit.getGIDs(std::string("Layer") + layer));
+    }
+    catch (...)
+    {
+        BRAYNS_ERROR << "Not all layer targets were found for neuron_by_layer"
+                        " color scheme"
+                     << std::endl;
+    }
+
+    std::vector<brion::GIDSet::const_iterator> iters;
+    for (const auto& layer : layers)
+        iters.push_back(layer.begin());
+
+    size_ts materialIds;
+    materialIds.reserve(gids.size());
+    // For each GID find in which layer GID list it appears, assign
+    // this index to its material id list.
+    for (auto gid : gids)
+    {
+        size_t i = 0;
+        for (i = 0; i != layers.size(); ++i)
+        {
+            while (iters[i] != layers[i].end() && *iters[i] < gid)
```

Comment:
Is there a reason why you store iterators and use while loops instead of using for-loop or std::find for each layer?

Reply:
Yes, based on the fact that GID sets are sorted, this code finds in which layer is present each gid by visiting exactly only once each element of every set. What you are proposing would be noticeably slower for milliions of cells (let's say O(n log n) vs O(n))

Reply:
Can you add the above comment in the code as well?

Reply:
Done